File: 0028-find-the-index-of-the-first-occurrence-in-a-string/0028-find-the-index-of-the-first-occurrence-in-a-string.go
func strStr(haystack string, needle string) int {
    if len(haystack) < len(needle){
        return -1
    }

    for i := range(len(haystack) - len(needle) + 1) {
        if haystack[i:i+len(needle)] == needle {
            return i
        }
    }

    return -1
        
}

File: 0028-find-the-index-of-the-first-occurrence-in-a-string/0028-find-the-index-of-the-first-occurrence-in-a-string.js
/**
 * @param {string} haystack
 * @param {string} needle
 * @return {number}
 */
var strStr = function(haystack, needle) {
    if (haystack.length < needle.length) return -1;

    for (let i = 0; i < haystack.length - needle.length + 1; i++) {
        if (haystack.slice(i, i + needle.length) === needle) return i;
    }

    return -1;
};